include/zephyr/logging/log_backend.h

```diff
@@ -341,6 +377,39 @@ static inline int log_backend_format_set(const struct log_backend *backend, uint
 }
 #endif
 
+/**
+ * @brief Notify a backend of an event.
+ *
+ * @param backend Pointer to the backend instance.
+ * @param event Event to be notified.
+ * @param arg Pointer to the argument(s).
+ */
+static inline void log_backend_notify(const struct log_backend *const backend,
+				      enum log_backend_evt event,
+				      union log_backend_evt_arg *arg)
+{
+#ifdef CONFIG_LOG_BACKEND_EVENT_NOTIFICATION_SUPPORT
+	__ASSERT_NO_MSG(backend != NULL);
+
+	if (backend->api->notify) {
+		backend->api->notify(backend, event, arg);
+	}
+#else
+	ARG_UNUSED(backend);
+	ARG_UNUSED(event);
+	ARG_UNUSED(arg);
+#endif
+}
+
+/**
+ * @brief Notify all backends of an event.
+ *
+ * @param event Event to be notified.
+ * @param arg Pointer to the argument(s).
+ */
+void log_backend_notify_all(enum log_backend_evt event,
```

Comment:
do we need it to be public? Maybe for now it can be static in `log_core.c` until there is a new feature request.

Reply:
Moved this into `log_core.c`.